### ISSUE TITLE
False negative result existed in SNP/InDel calling by samtools(v1.8) mpileup

### ISSUE THREAD

**Issue by wangce1988** [samtools/samtools#948]
Hello developers,

I'm using samtools-v1.8 on Linux-x86_64 compiled by gcc-4.8.5.
I was running samtools mpileup in SNP/InDel calling, and found a false negative SNP. This is a complex mutation with an insertion of TTA at chr17:37880996 and a SNP(G>T) at chr17:37880997, which can also be described as G>TTAT at chr17:37880997. 
However, mpileup detected the insertion but not detected the SNP. I have also tried the latest version of mpileup in bcftools-v1.9, and acquired similar results. The output near this location in vcf is as follows: 

17      37880996        .       T       <*>     0       .       DP=1431;I16=591,289,0,0,30695,1.07307e+06,0,0,43676,2.17137e+06,0,0,17978,413020,0,0;QS=1,0;MQSB=0.938328;MQ0F=0        PL:DP:SP:ADF:ADR:AD     0,255,255:880:0:591,0:289,0:880,0
17      37880996        .       T       TTTA    0       .       INDEL;IDV=373;IMF=0.260657;DP=1431;I16=595,448,209,163,77068,5.71977e+06,20252,1.12036e+06,51735,2.57096e+06,18505,921723,21478,493016,8434,200858;QS=0.741494,0.258506;VDB=0.301527;SGB=-0.693147;MQSB=0.920229;MQ0F=0 PL:DP:SP:ADF:ADR:AD     255,0,255:1415:1:595,209:448,163:1043,372
17      37880997        .       G       <*>     0       .       DP=1432;I16=571,418,0,0,34361,1.19626e+06,0,0,49062,2.43823e+06,0,0,20288,465166,0,0;QS=1,0;MQSB=0.929484;MQ0F=0        PL:DP:SP:ADF:ADR:AD     0,255,255:989:0:571,0:418,0:989,0
17      37880998        .       G       <*>     0       .       DP=1470;I16=578,444,0,0,35917,1.26427e+06,0,0,50668,2.51692e+06,0,0,20978,482268,0,0;QS=1,0;MQSB=0.875694;MQ0F=0        PL:DP:SP:ADF:ADR:AD     0,255,255:1022:0:578,0:444,0:1022,0

In IGV, this site(sorted by base) looks like:

![2018-09-12_16-24-33](https://user-images.githubusercontent.com/30822266/45412440-2f0e6180-b6a9-11e8-8105-eb4a108d12bf.png)

Please let me know if this is a bug or if I'm missing something. 
Thanks！

Ce Wang

**Comment by tseemann** [comment]
Can you please provide the *full set of commands you ran* to call the SNPs, including the read alignment step, please?

**Comment by wangce1988** [comment]
Sorry, I just forgot to provide my command line. 

Alignment step: 
bwa mem -R "@RG\tID:sampleID\tSM:sampleID\tPL:illumina\tDS:hg19\tCN:clgene" -t 8 -M hg19.fasta read1.fq.gz read2.fq.gz > sampleID.bam

Post alignment steps:
1. Sort by coordinate using Picard-v2.12.1
2. Target region filter by samtools view
3. Remove duplicate reads by in-house algorithm
4. Realigment by GATK-v3.8
Your attention please: 
I also tried just step 1 and 2 in post alignment steps. Finally, acquired similar results.

SNPs/InDels calling:
samtools mpileup -u -v -E -C 50 -q 30 -Q 30 -t DP,AD,ADF,ADR,SP -f hg19.fasta -d 100000 -L 100000 -F 0.0005 -l target.bed sampleID.rmdup.realign.bam >sampleID.snvIndel.raw.vcf
Your attention please: 
Base quality of altered "T" at this position is high enough for calling.

**Comment by tseemann** [comment]
In IGV are you filtering the reads you can see by q=30 and Q=30 and any soft-clipping in the alignments?

Otherwise you are not comparing the same things.

**Comment by wangce1988** [comment]
I have checked the "q" and "Q" of so many altered bases by mouse selection of each. Almost all values are higher than 30. The values can be shown like this:
![408863508](https://user-images.githubusercontent.com/30822266/45470770-9556bb00-b761-11e8-998a-29e51102801b.jpg)

I have also tried the latest version of mpileup in bcftools-v1.9, but acquired similar results as I mentioned before.

"-v" tells samtools mpileup to output VCF format in v1.8.
![_2018-09-13_14-35-19](https://user-images.githubusercontent.com/30822266/45471103-9dfbc100-b762-11e8-968c-e798b11a2d8a.png)

What I analyzed are ctDNA samples with very low mutation frequency. Other callers do not have good performance. So I use samtools mpileup to convert bam to vcf, then filter mutations by script.
